gis/text_to_centreline_geometry_scripts/README.md

```diff
@@ -0,0 +1,189 @@
+
+The type of buffer we are using in both cases has a flat endcap. This means that that the buffer does not go farther than the end of `line_geom`. This type of buffer is especially useful in the case where `metres_btwn1` and `metres_btwn2` are `NULL` because it prevents additional centreline segments that are part of the same street and are beside each intersection (but not inside the buffer) from being included in the selected centreline segments. It is also useful in the case where `metres_btwn1` and/or `metres_btwn2` are not `NULL` (where we effectively add 30 metres to the end of each buffer), because if the endcap was not flat then the length from the intersection to the end of the buffer would be a function on the length of the buffer, and we would not easily know this value/it could be too long if the buffer is long. 
+
+### centreline_segments variable:
+
+In the `text_to_centreline` function, this variable is the final version of the geometry that represents the location between the two intersections. There are a few cases for how to get the final geometry from the `match_line_to_centreline` geometry:
+
+1. If the bylaw occurs on the entire length of the street then call a special function called `get_entire_length_centreline_segements`. This function selects all centreline segments in the City of Toronto with the exact name of `highway2`. In this case the street name has to be exact, and if the street name is misspelled then there will be no output geometry. There could potentially be an issue with how this code is written because some streets in the city have the same name but occur in different districts (i.e. Etobicoke and North York).
+
+2. If `COALESCE(metres_btwn1, metres_btwn2) IS NULL` then the bylaw is not a [special case](#Special-Cases). So the centreline segments from `match_line_to_centreline` are the final output.
+
+3. If the bylaw is [special case](#Special-Cases) the call the appropriate function for that special case.
+
+
+### Confidence output
+
+The function outputs a confidence level (`con`), which is defined after the `centreline_segments` variable. This value represents how close the names of the input values are to the intersections they were matched to. Specifically, this value is the sum of the levenshtien distance between the `highway2` value and the street name it matched to and the `btwn1`/`btwn2` value and the street name it matched to. It is the sum of this value for both intersections.
+
+# Special Cases
+
+There are some records that start and/or end at locations that are a certain amount of metres away from an intersection. We create a method to assign a geometry to the locations of these bylaws. These segments do not start at one intersection and end at another, they often start/end in the middle of centreline segments.
+
+### Explore the data/cases
+
+**Case 1**
```

Comment:
The cases can be summarized as:
- "Two intersections"
- "One Intersection + offset"
- "Two intersections and at least one offset"
- "Offset streets" (the multi-linestring issue).